beacon/sync/src/main/java/tech/pegasys/teku/beacon/sync/historical/ReconstructHistoricalStatesService.java

```diff
@@ -93,12 +102,33 @@ protected SafeFuture<?> doStart() {
 
   public void applyBlocks(final BeaconState genesisState, final UInt64 anchorSlot) {
     Context context = new Context(genesisState, SpecConfig.GENESIS_SLOT.plus(1), anchorSlot);
-    applyNextBlock(context).finish(statusLogger::reconstructHistoricalStatesServiceFailedProcess);
+    applyNextBlock(context)
+        .finish(
+            error -> {
+              final Throwable rootCause = Throwables.getRootCause(error);
+              if (rootCause instanceof ShuttingDownException
+                  || rootCause instanceof InterruptedException
+                  || rootCause instanceof RejectedExecutionException) {
+                LOG.debug("Shutting down", rootCause);
+              } else {
+                statusLogger.reconstructHistoricalStatesServiceFailedProcess(error);
+              }
+            });
   }
 
   private SafeFuture<Void> applyNextBlock(Context context) {
     if (context.checkStopApplyBlock()) {
       statusLogger.reconstructHistoricalStatesServiceComplete();
+      stopped.complete(null); // fixme should this be set here too?
+      return SafeFuture.COMPLETE;
+    }
+
+    if (shutDown.get()) {
+      statusLogger.reconstructHistoricalStatesServiceFailedProcess( // fixme log out stop?
+          new FatalServiceFailureException(
+              "ReconstructHistoricalStatesService", "Service was stopped before complete"));
```

Comment:
We don't need to log anything here - teku is just shutting down and we'll resume reconstructing states when we start up again.